www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NginxSetIfConstraint.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ *    Copyright (C) 2018 Fabian Franz
+ *    Copyright (C) 2018 Deciso B.V.
+ *
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *    THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ *    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+namespace OPNsense\Base\Constraints;
+
+/**
+ * validate if a field is set depening on the setting of another field
+ * Class NginxSetIfConstraint
+ * @package OPNsense\Base\Constraints
+ */
+class NginxSetIfConstraint extends BaseConstraint
```

Comment:
@fichtner you may want to move that to core since it is very generic and can be used by others as well.

Reply:
isn't there something similar already? I don't mind core inclusion, but up to @AdSchellevis

Reply:
@fichtner there is one which checks if another value exists this one does check for equality (used for "OptionField" values if some are required if a specific value is selected. You can remove the prefix and move it to core but at the moment it is only strictly required in this plugin.